evm/db/backends/base.py

```diff
@@ -12,15 +19,24 @@ def __init__(self) -> None:
             "The `init` method must be implemented by subclasses of BaseDB"
         )
 
-    @abstractmethod
-    def get(self, key: bytes) -> bytes:
-        """Return the value for the given key.
+    def get(self, key: bytes, default: Union[bytes, Unset] = unset) -> bytes:
+        """
+        Return the value for the given key.
+
+        If the key doesn't exist, and a default is provided, return the default value.
+        If the key doesn't exist, and a default is not provided, raise a KeyError
 
-        Raises KeyError if key doesn't exist.
+        :return: the value with the associated key
+        :raise: KeyError if key is missing
         """
-        raise NotImplementedError(
-            "The `get` method must be implemented by subclasses of BaseDB"
-        )
+        try:
+            return self[key]
+        except KeyError as exc:
+            if default is unset:
+                raise exc
```

Comment:
Anyone have a preference for re-raising.  I've always used a bare `raise` like this (which looks like it'd allow you to remove the `as exc` part.

```python
try:
    thing()
except SomeError:
    raise
```